### ISSUE TITLE
CISM fields needed from data ocean for antarctic coupling

### ISSUE THREAD

**Issue by mvertens** [ESCOMP/CMEPS#453]
In CISM, subroutine glad_gcm (in glad_main.F90) is set up to receive two 3D ocean fields on the CISM grid:

real(dp),dimension(:,:,:),intent(in)  :: salinity     ! input ocean salinity (g/kg)                                                  
real(dp),dimension(:,:,:),intent(in)  :: tocn         ! input ocean temperature (deg K)   

The first dimension is the ocean level, and the second and third are the horizontal dimensions. CISM also has an internal ocean_data derived type that includes an integer nzocn (the number of ocean levels) and a 1D array zocn (the ocean elevation in 

Should the input ocean temperature should be the potential temperature rather than the in situ temperature?

To compute sub-shelf melting internally, CISM needs a 3D variable called thermal_forcing (TF), which depends on ocean T, S, and depth. Each time glad_gcm is called, CISM uses T and S at each level to compute TF. The resulting TF is accumulated and averaged over a mass balance time step, like the SMB and temperature fields (qsmb and tsfc) from the land model.

In addition the mediator also needs the ocean masks (i.e. bathymetry) at the sub-grid levels passed.
So in short, a data ocean model should pass 3D T, S and mask fields to the mediator. The mediator should regrid these fields to the CISM grid(s), inserting a fill value where the ocean and ice-sheet grids don't overlap. And at some point during initialization, CISM needs to be told how many ocean levels there are and where they're located.

@whlipscomb - please add more details that you think are necessary. Can you also point me to a dataset that would have these fields so that I can start prototyping this.
